cluster/store/query.go

```diff
@@ -0,0 +1,64 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+
+	cmd "github.com/weaviate/weaviate/cluster/proto/cluster"
+)
+
+func (st *Store) Query(req *cmd.QueryRequest) (*cmd.QueryResponse, error) {
+	st.log.Debug("server.query", "type", req.Type)
+
+	var payload []byte
+	var err error
+	switch req.Type {
+	case cmd.QueryRequest_TYPE_GET_READONLY_CLASS:
+		payload, err = st.QueryReadOnlyClass(req)
+		if err != nil {
+			return nil, fmt.Errorf("could not get read only class: %w", err)
+		}
+	default:
+		// This could occur when a new command has been introduced in a later app version
+		// At this point, we need to panic so that the app undergo an upgrade during restart
+		const msg = "consider upgrading to newer version"
```

Comment:
I believe it would be better not to panic at this stage and return an error so the client can retry the request if needed. The read request does not justify the panic in my opinion, especially considering that it will crash the leader itself. While I understand the need for software upgrade, they should not come at the cost of system reliability

Reply:
I'll make it return an error that should get propagated back to the client, so at least client will be aware of the need to upgrade.

Do note that this behaviour i got from the `Execute` from schema writes we have. Maybe we also don't want to panic on schema write with an old version ? Just a suggestion.